src/main/java/io/fabric8/vertx/maven/plugin/mojos/AbstractRunMojo.java

```diff
@@ -423,14 +422,22 @@ private boolean isYaml(String configFile) {
 
         @Override
         public void call(String kind, Path path) {
+
             final MojoUtils mojoUtils = new MojoUtils().withLog(getLog());
 
             if (getLog().isDebugEnabled()) {
                 getLog().debug(" Received Build request with kind:" + kind + " for path " + path);
             }
 
             try {
-                mojoUtils.compile(project, mavenSession, buildPluginManager);
+
+                //Anything other than java will be treated as resources
+                if ("java".equals(org.codehaus.plexus.util.FileUtils.extension(path.toString()))) {
```

Comment:
Actually I would avoid this check here by using _2_ `IncrementalBuilder` with two different callbacks: One for the source and one for the resources. This makes things also more direct to have a `CompileCallback` and `CopyResourceCallback`. In `FileUtils` you would then have to different methods for detecting resources and java files.